a5py/ascot5io/wall.py

```diff
@@ -235,7 +240,17 @@ def noderepresentation(self):
                 define the indices of the vertices in the vertices array.
         """
         with self as h5:
-            ntriangle = int(h5["nelements"][:])
+            if filter_wall is None:
```

Comment:
How about `filter_wall = np.s_[:] if filter wall is None` or something like that and then you can use this for both cases:
```
                x1x2x3 = h5["x1x2x3"][filter_wall, :]
                y1y2y3 = h5["y1y2y3"][filter_wall, :]
                z1z2z3 = h5["z1z2z3"][filter_wall, :]
                ntriangle = x1x2x3.shape[0]
```